client/data/followers/use-followers.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { useInfiniteQuery } from 'react-query';
+import { uniqueBy } from '@automattic/js-utils';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const MAX_FOLLOWERS = 10; // means pages (= 1000 followers);
+
+/**
+ * Normalizes a follower object. Changes 'avatar' to 'avatar_URL' allowing a follower
+ * object to be used with the Gravatar component.
+ *
+ * @param  {object} follower A follower ojbect
+ * @returns {object}          A normalized follower object
+ */
+export function normalizeFollower( follower ) {
+	follower.avatar_URL = follower.avatar;
+	return follower;
+}
+
+const extractPages = ( pages = [] ) =>
+	pages.flatMap( ( page ) => page.subscribers ).map( normalizeFollower );
+const compareUnique = ( a, b ) => a.ID === b.ID;
+
+const useFollowers = ( siteId, fetchOptions = {}, queryOptions = {} ) => {
+	const { search, type } = fetchOptions;
+
+	return useInfiniteQuery(
+		[ 'followers', siteId, type, search ],
+		async ( { pageParam = 1 } ) => {
+			const res = await wpcom.site( siteId ).statsFollowers( { ...fetchOptions, page: pageParam } );
```

Comment:
It's interesting that the `.statsFollowers()` method calls the `/sites/:site/stats/followers` endpoint, while the original data layer handler calls `/sites/:site/followers`.

They both return exactly the same results. The only difference is that the `stats` one does an additional check testing if the Stats module is enabled on a Jetpack site. On Simple sites there's no difference at all.

For this UI that's outside Stats the `/followers` endpoint looks like the better one.

Reply:
Good point! Was about to change it anyway to use `wpcom.req.get` similar to https://github.com/Automattic/wp-calypso/pull/50712#discussion_r587311182.